compiler/rustc_mir/src/const_eval/mod.rs

```diff
@@ -35,6 +38,97 @@ pub(crate) fn const_caller_location(
     ConstValue::Scalar(loc_place.ptr)
 }
 
+/// Convert an evaluated constant to a type level constant
+pub(crate) fn const_to_valtree<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    param_env: ty::ParamEnv<'tcx>,
+    raw: ConstAlloc<'tcx>,
+) -> Option<ty::ValTree<'tcx>> {
+    let ecx = mk_eval_cx(tcx, DUMMY_SP, param_env, false);
```

Comment:
Please add a comment saying that the `false` here is *absolutely crucial for soundness* since it makes sure that all our reads will be from read-only memory.